Sources/Datadog/Core/Upload/DataUploadWorker.swift

```diff
@@ -67,21 +67,23 @@ internal class DataUploadWorker: DataUploadWorkerType {
 
                 if shouldBeAccepted {
                     self.fileReader.markBatchAsRead(batch)
+                    self.delay.decrease()
+
                     developerLogger?.info("   → (\(self.featureName)) accepted, won't be retransmitted: \(uploadStatus)")
                     userLogger.debug("   → (\(self.featureName)) accepted, won't be retransmitted: \(uploadStatus)")
                 } else {
+                    self.delay.increase()
+
                     developerLogger?.info("  → (\(self.featureName)) not delivered, will be retransmitted: \(uploadStatus)")
                     userLogger.debug("   → (\(self.featureName)) not delivered, will be retransmitted: \(uploadStatus)")
                 }
-
-                self.delay.decrease()
             } else {
                 let batchLabel = nextBatch != nil ? "YES" : (isSystemReady ? "NO" : "NOT CHECKED")
                 let systemLabel = isSystemReady ? "✅" : "❌"
                 developerLogger?.info("💡 (\(self.featureName)) No upload. Batch to upload: \(batchLabel), System conditions: \(systemLabel)")
                 userLogger.debug("💡 (\(self.featureName)) No upload. Batch to upload: \(batchLabel), System conditions: \(systemLabel)")
 
-                self.delay.increaseOnce()
+                self.delay.increase()
```

Comment:
i couldn't find a way to test `increase/decrease` logic here (without sleeping the thread for a while)
i'm thinking of moving this logic to `UploadDelay` type, basically like that:
```swift
struct DataUploadDelay {
  ...
  mutating func handle(uploadStatus: DataUploadStatus?) {
    if nil -> no upload tried (increase)
    if failure -> upload failed (increase)
    if success/droppable -> upload succeeded (decrease)
  }
}
```
then i can test the logic without `DataUploadWorker`/`queue.async {...}`
otherwise i can leave it as is, we don't have tests for this logic already

wdyt @ncreated ?